flutter/android/src/main/kotlin/io/sentry/flutter/SentryFlutterPlugin.kt

```diff
@@ -142,6 +178,41 @@ class SentryFlutterPlugin : FlutterPlugin, MethodCallHandler {
     }
   }
 
+  private fun beginNativeFrames(result: Result) {
+    activity?.let {
+      activityFramesTracker.addActivity(it)
+    }
+    result.success(null)
+  }
+
+  private fun endNativeFrames(id: String?, result: Result) {
+    if (id == null) {
+      Log.w("Sentry", "Parameter id cannot be null.")
+      result.success(null)
+      return
+    }
+    val sentryId = SentryId(id)
+    activity?.let {
+      activityFramesTracker.setMetrics(it, sentryId)
+
+      val metrics = activityFramesTracker.takeMetrics(sentryId)
+      val total = metrics?.get("frames_total")?.getValue()
+      val slow = metrics?.get("frames_slow")?.getValue()
+      val frozen = metrics?.get("frames_frozen")?.getValue()
+
+      if (total == null || slow == null || frozen == null) {
```

Comment:
You can check if `metrics` is null only, otherwise everything else would have a value

Reply:
Compiler will still assume nullability if we don't check the individual variables, as it we are pulling them out of a dictionary.